### PR TITLE
Expand test coverage; no longer featurize in Docker; use symlinks so Disco workers can run Docker

### PR DESCRIPTION
**Pull request by acrellin**

### CODE REVIEW

File: mltsp/Flask/flask_app.py [hunk 1]
```diff
@@ -2015,25 +2011,26 @@ def prediction_proc(
...
+                                                                         i + 1)
     results_str += (
         "        </tr>"
         "    </thead>"
         "    <tbody>")
     try:
-        results_dict = run_in_docker_container.predict_in_docker_container(
-            newpred_file_path, model_name, model_type,
-            prediction_entry_key, featset_key, sep=sep,
+        # results_dict = run_in_docker_container.predict_in_docker_container(
```
Comment (stefanv) [inline mltsp/Flask/flask_app.py]: Some commented code left

File: mltsp/Flask/flask_app.py [hunk 2]
```diff
@@ -1881,14 +1877,14 @@ def featurize_proc(
...
     try:
-        results_str = run_in_docker_container.featurize_in_docker_container(
-            headerfile_path, zipfile_path, features_to_use, featureset_key,
-            is_test, already_featurized, custom_script_path)
-        # results_str = featurize.featurize(
-        #    headerfile_path, zipfile_path, features_to_use=features_to_use,
-        #    featureset_id=featureset_key, is_test=is_test,
-        #    already_featurized=already_featurized,
-        #    custom_script_path=custom_script_path, USE_DISCO=False)
+        # results_str = run_in_docker_container.featurize_in_docker_container(
```
Comment (stefanv) [inline mltsp/Flask/flask_app.py]: Here too

File: mltsp/Flask/flask_app.py [hunk 3]
```diff
@@ -2015,25 +2011,26 @@ def prediction_proc(
...
+        #     custom_features_script=custom_features_script)
+
+        results_dict = predict.predict(
+            newpred_file_path=newpred_file_path, model_name=model_name,
+            model_type=model_type, featset_key=featset_key, sepr=sep,
             n_cols_html_table=n_cols_html_table,
-            features_already_extracted=None, metadata_file=metadata_file,
-            custom_features_script=custom_features_script)
-
-        # results_dict = predict.predict(
```
Comment (stefanv) [inline mltsp/Flask/flask_app.py]: ...

File: mltsp/custom_feature_tools.py [hunk 4]
```diff
@@ -482,7 +492,7 @@ def remove_tmp_files(path_to_tmp_dir):
         try:
             os.remove(tmp_file)
         except Exception as e:
```
Comment (stefanv) [inline mltsp/custom_feature_tools.py]: Ideally, catch the exact Exception, e.g. ``except IOError: pass``

File: mltsp/cfg.py [hunk 5]
```diff
@@ -27,6 +27,8 @@
 TCP_INGEST_TOOLS_PATH = os.path.join(MLTSP_PACKAGE_PATH,
                                      "TCP/Software/ingest_tools")
 
+PROJECT_PATH_LINK = "/tmp/mltsp_link"
```
Comment (stefanv) [inline mltsp/cfg.py]: What is "PROJECT_PATH_LINK" for?

File: mltsp/custom_feature_tools.py [hunk 6]
```diff
@@ -460,12 +470,12 @@ def extract_feats_in_docker_container(container_name, path_to_tmp_dir):
         with open(os.path.join(path_to_tmp_dir, "results_list_of_dict.pkl"),
                   "rb") as f:
             results_list_of_dict = pickle.load(f)
+            return results_list_of_dict
     except:
```
Comment (stefanv) [inline mltsp/custom_feature_tools.py]: ``except raise`` is probably the same as not catching the error?

File: mltsp/util.py [hunk 7]
```diff
@@ -0,0 +1,25 @@
+import subprocess
```
Comment (stefanv) [inline mltsp/util.py]: This already exists as ``docker_util``, so we should merge those two submodules.

**Comment by stefanv** [comment]
It's green!  I left some minor feedback

File: mltsp/custom_feature_tools.py (hunk 6 above)
Comment (acrellin) [inline mltsp/custom_feature_tools.py]: It's all about the `finally` block!

Comment (acrellin) [inline mltsp/custom_feature_tools.py]: `try: 
    ...
except:
    raise
finally:
   # I want this to happen even if an exception is raised in the try block`

Comment (stefanv) [inline mltsp/custom_feature_tools.py]: Ah, I see! ``try: finally:`` can be used without the except, though

**Comment by stefanv** [comment]
In it goes!

**Pull request merged**